attestation-service/verifier/src/snp/mod.rs

```diff
@@ -56,9 +56,18 @@ impl Verifier for Snp {
             return Err(anyhow!("VMPL Check Failed"));
         }
 
-        let expected_report_data = calculate_expected_report_data(&nonce, &attestation.tee_pubkey);
-        if report.report_data != expected_report_data {
-            return Err(anyhow!("Report Data Mismatch"));
+        if let Some(expected_report_data) = expected_report_data {
+            debug!("Check the binding of REPORT_DATA.");
+            if expected_report_data != report.report_data {
+                bail!("Report Data Mismatch");
+            }
+        }
```

Comment:
I don't think this is the same behaviour. `report_data` is now optional for this verifier, afaik this is not what we want.

```suggestion
        let Some(expected_report_data) = expected_report_data else {
          bail!("Report Data unset");
        };  
        debug!("Check the binding of REPORT_DATA.");
        if expected_report_data != report.report_data {
          bail!("Report Data Mismatch");
        }
```


Arguably `Option<&[u8]>` might not be the right type for report_data. It might make sense to have a dedicated, expressive enum to prevent misuse.

```rust
#[must_use]
enum ReportData<'a> {
   Value(&'a[u8]),
   Unsupported,
}

let ReportData::Value(val) = expected_report_data else {
   return MyError("no report data!");
}   
```

Reply:
Thanks for the suggestions. Let me fix